ga4gh/datamodel/__init__.py

```diff
@@ -229,6 +256,8 @@ class VariantSetCompoundId(DatasetCompoundId):
     """
     fields = DatasetCompoundId.fields + ['variantSet']
     containerIds = DatasetCompoundId.containerIds + [('variantSetId', 1)]
+    obfuscators = [DatasetCompoundId, None]
```

Comment:
The compoundId object often does contain the ids of other objects within it (for instance, a ReadGroup compoundId contains the ReadGroupSetId of which it belongs to).  In order for the, for instance, readGroupCompoundId.readGroupSetId to match the original readGroupSet id, they must be obfuscated in the same way.